File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.java
class Solution {
    public int maxProfit(int[] prices) {
        int left=0;
        int n=prices.length;
        int maxProfit=0;

        for(int right=1;right<n;right++){
            //cheaper day to buy
            if(prices[right]<prices[left]){
                left=right;
            }
            else{
                int profit=prices[right]-prices[left];
                maxProfit=Math.max(maxProfit, profit);
            }
            
        }
        return maxProfit;
        
    }
}

File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        left=0
        right=len(heights)-1
        maxArea=0


        while left<right:
            maxArea=max(maxArea,((right-left)*(min(heights[left],heights[right]))))

            if heights[left]>=heights[right]:
                right-=1
            else:
                left+=1
        
        return maxArea




            


        
        